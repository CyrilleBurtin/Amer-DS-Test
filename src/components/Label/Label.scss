.slm-label {
  @apply rounded-3xl
    px-2
    py-0.5;

  &.--gray {
    @apply bg-salomon-gray-200
      text-salomon-gray-400;
  }

  &.--red {
    @apply bg-salomon-red
      text-salomon-gray-200;
  }
}
