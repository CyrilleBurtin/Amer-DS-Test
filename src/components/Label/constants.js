export const LABEL_VARIANT = {
  GRAY: 'gray'
}
