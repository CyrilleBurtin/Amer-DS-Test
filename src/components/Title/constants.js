export const TITLE_VARIANT = {
  RED: 'red',
  GRAY: 'gray'
};
