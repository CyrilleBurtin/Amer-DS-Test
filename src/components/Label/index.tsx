import { type Label } from '../../interfaces/label'
import './Label.scss'
import { classNames } from '../../utils'
import './Label.scss'

const Label = ({ variant = 'gray', children }: Label) => {
  const classes = {
    label: classNames('slm-label', `--${variant}`)
  }

  return <span className={classes.label}>{children}</span>
}

export default Label
