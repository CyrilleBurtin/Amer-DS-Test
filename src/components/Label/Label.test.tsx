import { render } from '@testing-library/react'
import Label from './index.js'
import { LABEL_VARIANT } from './constants.js'

describe('<Label>', () => {
  it('should render the default component (gray)', () => {
    const { container, getByText } = render(<Label>No variant passed</Label>)

    expect(getByText(/No variant passed/i)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should render the gray variant component with custom text', () => {
    const { container, getByText } = render(
      <Label variant={LABEL_VARIANT.GRAY}>Label content</Label>
    )

    expect(getByText(/Label content/i)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should render the red variant component with custom text', () => {
    const { container, getByText } = render(
      <Label variant={LABEL_VARIANT.RED}>Red variant</Label>
    )

    expect(getByText(/Red variant/i)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
