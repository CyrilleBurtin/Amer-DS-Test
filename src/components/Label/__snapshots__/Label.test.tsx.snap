// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Label> > should render the default component (gray) 1`] = `
<div>
  <span
    class="slm-label --gray"
  >
    No variant passed
  </span>
</div>
`;

exports[`<Label> > should render the default component 1`] = `
<div>
  <span
    class="slm-label --gray"
  >
    No variant passed
  </span>
</div>
`;

exports[`<Label> > should render the gray variant component with custom text 1`] = `
<div>
  <span
    class="slm-label --gray"
  >
    Label content
  </span>
</div>
`;

exports[`<Label> > should render the red variant component with custom text 1`] = `
<div>
  <span
    class="slm-label --red"
  >
    Red variant
  </span>
</div>
`;
